torch/utils/cpp_extension.py

```diff
@@ -116,37 +133,31 @@ def check_compiler_abi_compatibility(compiler):
         return True
     if os.environ.get('TORCH_DONT_CHECK_COMPILER_ABI') in ['ON', '1', 'YES', 'TRUE', 'Y']:
         return True
-    try:
-        check_cmd = '{}' if IS_WINDOWS else '{} --version'
-        info = subprocess.check_output(
-            check_cmd.format(compiler).split(), stderr=subprocess.STDOUT)
-    except Exception:
-        _, error, _ = sys.exc_info()
-        warnings.warn('Error checking compiler version: {}'.format(error))
-    else:
-        info = info.decode().lower()
-        if 'gcc' in info or 'g++' in info:
-            # Sometimes the version is given as "major.x" instead of semver.
-            version = re.search(r'(\d+)\.(\d+|x)', info)
-            if version is not None:
-                major, minor = version.groups()
-                minor = 0 if minor == 'x' else int(minor)
-                if (int(major), minor) >= MINIMUM_GCC_VERSION:
-                    return True
-                else:
-                    # Append the detected version for the warning.
-                    compiler = '{} {}'.format(compiler, version.group(0))
-        elif 'Microsoft' in info:
-            info = info.decode().lower()
-            version = re.search(r'(\d+)\.(\d+)\.(\d+)', info)
-            if version is not None:
-                major, minor, revision = version.groups()
-                if (int(major), int(minor),
-                        int(revision)) >= MINIMUM_MSVC_VERSION:
-                    return True
-                else:
-                    # Append the detected version for the warning.
-                    compiler = '{} {}'.format(compiler, version.group(0))
+
+    # First check if the compiler is one of the expected ones for the particular platform.
+    if IS_WINDOWS or check_compiler_ok_for_platform(compiler):
```

Comment:
Maybe print a different error if the compiler is not ok for the platform than if the version is wrong? Or reflect in the warning what is expected here?
In particular, for mac, if you use any gcc at the moment, it will just tell you to use gcc 4.9+ in the warning. But no gcc will actually work.